File: python/1614.maximum_nesting_depth_of_the_parentheses.py
class Solution:
    def maxDepth(self, s):
        count = 0
        max_count = 0
        for i in s:
            if i == "(":
                count += 1
            elif i == ")":
                count -= 1
            max_count = max(max_count, count)
        return max_count

File: cpp/1614.maximum_nesting_depth_of_the_parentheses.cpp
#include <string>
class Solution {
 public:
    int maxDepth(std::string s) {
        int count = 0;
        int result = 0;

        for (auto ch : s) {
            if (ch == '(') {
                ++count;
            } else if (ch == ')') {
                --count;
            }
            result = std::max(count, result);
        }
        return result;
    }
};
